plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/bitbucketServerPullRequest.ts

```diff
@@ -0,0 +1,266 @@
+import { parseRepoUrl } from './util';
+import { Config } from '@backstage/config';
+
+const createPullRequest = async (opts: {
+  project: string;
+  repo: string;
+  title: string;
+  description?: string;
+  toRef: object;
+  fromRef: object;
+  authorization: string;
+  apiBaseUrl: string;
+}) => {
+  const {
+    project,
+    repo,
+    title,
+    description,
+    toRef,
+    fromRef,
+    authorization,
+    apiBaseUrl,
+  } = opts;
+
+  let response: Response;
+  const data: RequestInit = {
+    method: 'POST',
+    body: JSON.stringify({
+      title: title,
+      description: description,
+      state: 'OPEN',
+      open: true,
+      closed: false,
+      locked: true,
+      toRef: toRef,
+      fromRef: fromRef,
+    }),
+    headers: {
+      Authorization: authorization,
+      'Content-Type': 'application/json',
+    },
+  };
+
+  try {
+    response = await fetch(
+      `${apiBaseUrl}/projects/${project}/repos/${repo}/pull-requests`,
+      data,
+    );
+  } catch (e) {
+    throw new Error(`Unable to create pull-reqeusts, ${e}`);
+  }
+
+  if (response.status !== 201) {
+    throw new Error(
+      `Unable to create pull requests, ${response.status} ${
+        response.statusText
+      }, ${await response.text()}`,
+    );
+  }
+
+  const r = await response.json();
+  return `${r.links.self[0].href}`;
+};
+const findBranches = async (opts: {
+  project: string;
+  repo: string;
+  branchName: string;
+  authorization: string;
+  apiBaseUrl: string;
+}) => {
+  const { project, repo, branchName, authorization, apiBaseUrl } = opts;
+
+  let response: Response;
+  const options: RequestInit = {
+    method: 'GET',
+    headers: {
+      Authorization: authorization,
+      'Content-Type': 'application/json',
+    },
+  };
+
+  try {
+    response = await fetch(
+      `${apiBaseUrl}/projects/${project}/repos/${repo}/branches?boostMatches=true&filterText=${branchName}`,
+      options,
+    );
+  } catch (e) {
+    throw new Error(`Unable to get branches, ${e}`);
+  }
+
+  if (response.status !== 200) {
+    throw new Error(
+      `Unable to get branches, ${response.status} ${
+        response.statusText
+      }, ${await response.text()}`,
+    );
+  }
+
+  const r = await response.json();
+  let branch;
+  for (const object of r.values) {
+    if (object.displayId === branchName) {
+      branch = object;
+      break;
+    }
+  }
+
+  return branch;
+};
+
+/**
+ * Creates a BitbucketServer Pull Request action.
+ * @public
+ */
+export function createPublishBitbucketServerPullRequestAction(options: {
+  integrations: ScmIntegrationRegistry;
+  config: Config;
+}) {
+  const { integrations } = options;
+
+  return createTemplateAction<{
+    repoUrl: string;
+    title: string;
+    description?: string;
+    targetBranch: string;
+    sourceBranch: string;
+    token?: string;
+  }>({
+    id: 'publish:bitbucketServer:pull-request',
+    schema: {
+      input: {
+        type: 'object',
+        required: ['repoUrl'],
```

Comment:
Seems from the types that more than this is required, right? title, branches etc